From 903d6fc2fc8a050caceb6f95363f6cc4572b9fc3 Mon Sep 17 00:00:00 2001
From: Derek Higgins <derekh@redhat.com>
Date: Thu, 26 Oct 2017 15:29:42 +0100
Subject: [PATCH] Skip iscsi attachment if no info

If no iscsi session information can be retrieved from the firmware
then skip the iscsi attachment and allow the boot process to continue.
Ensure the timeout scripts don't hit their timeout waiting for
/tmp/iscsistarted-firmware to be created.
This will allow a common image to be used for servers with both a
local and iscsi root with rd.iscsi.firmware set.

Cherry-picked from: 9408a2ac330970e98dc4d289ec3ffb163f0517d1
Resolves: #1513397
---
 modules.d/95iscsi/iscsiroot.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/modules.d/95iscsi/iscsiroot.sh b/modules.d/95iscsi/iscsiroot.sh
index 4cf3d355..c93ff068 100755
--- a/modules.d/95iscsi/iscsiroot.sh
+++ b/modules.d/95iscsi/iscsiroot.sh
@@ -46,8 +46,9 @@ fi
 handle_firmware()
 {
     if ! iscsistart -f; then
-        warn "iscistart: Could not get list of targets from firmware."
-        return 1
+        warn "iscistart: Could not get list of targets from firmware. Skipping."
+        echo 'skipped' > "/tmp/iscsistarted-firmware"
+        return 0
     fi
 
     for p in $(getargs rd.iscsi.param -d iscsi_param); do
